Podcast/SearchDiscoverViewController.swift

```diff
@@ -343,30 +334,36 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
     }
 
     func didPressFollowButton(cell: SearchPeopleTableViewCell) {
-        guard let data = searchResultsTableView.dataSource as? SearchDataSourceDelegate, let indexPath = searchResultsTableView.indexPath(for: cell), let user = data.searchResults[indexPath.row] as? User else { return }
+        guard let data = searchResultsTableView.dataSource as? MainSearchDataSourceDelegate, let indexPath = searchResultsTableView.indexPath(for: cell), let user = data.searchResults[.people]![indexPath.row] as? User else { return }
         user.followChange(completion: cell.setFollowButtonState)
     }
 
     func didPressSubscribeButton(cell: SearchSeriesTableViewCell) {
-        guard let data = searchResultsTableView.dataSource as? SearchDataSourceDelegate, let indexPath = searchResultsTableView.indexPath(for: cell), let series = data.searchResults[indexPath.row] as? Series else { return }
+        guard let data = searchResultsTableView.dataSource as? MainSearchDataSourceDelegate, let indexPath = searchResultsTableView.indexPath(for: cell), let series = data.searchResults[.series]![indexPath.row] as? Series else { return }
         series.subscriptionChange(completion: cell.setSubscribeButtonToState)
     }
 
     func didPressPlayButton(cell: SearchEpisodeTableViewCell) {
-        guard let data = searchResultsTableView.dataSource as? SearchDataSourceDelegate, let indexPath = searchResultsTableView.indexPath(for: cell), let appDelegate = UIApplication.shared.delegate as? AppDelegate, let episode = data.searchResults[indexPath.row] as? Episode else { return }
+        guard let data = searchResultsTableView.dataSource as? MainSearchDataSourceDelegate, let indexPath = searchResultsTableView.indexPath(for: cell), let appDelegate = UIApplication.shared.delegate as? AppDelegate, let episode = data.searchResults[.episodes]![indexPath.row] as? Episode else { return }
```

Comment:
can you safely unwrap the `data.searchResults[.episodes]`? just in case
although it seems like you have to force unwrap everywhere... is there a better way to handle that for the dictionary?